### PR TITLE
httpd: added ability to configure different credentials per listener

### PR DESCRIPTION
**Pull request by mmaslankaprv**
For some deployments it may be required to configure different set of
credentials for different http server listeners. Added ability to
set/override default http server credentials with listener specific set
of credentials.

Signed-off-by: Michal Maslanka <michal@vectorized.io>

### CODE REVIEW

File: src/http/httpd.cc [hunk 1]
```diff
@@ -473,14 +473,31 @@ void http_server::set_content_streaming(bool b) {
     _content_streaming = b;
 }
 
-future<> http_server::listen(socket_address addr, listen_options lo) {
-    if (_credentials) {
-        _listeners.push_back(seastar::tls::listen(_credentials, addr, lo));
+future<> http_server::listen(socket_address addr, listen_options lo, 
+            shared_ptr<seastar::tls::server_credentials> listener_credentials) {
+    if(listener_credentials) {
```
Comment (BenPope) [inline src/http/httpd.cc:478]: I wonder if the semantics should be that a null `listener_credentials` disables the default `_credentials`?

Comment (mmaslankaprv) [reply src/http/httpd.cc:478]: good point, we should probably discuss this, it would require a little bit more code

**Review by BenPope — APPROVED** [review]
LGTM.

Since they have: `listen_options lo`
I wonder if it shouldn't be: `shared_ptr<seastar::tls::server_credentials> lc`
But we can leave that to them to decide.

**Comment by mmaslankaprv** [comment]
> LGTM.
> 
> Since they have: `listen_options lo`
> I wonder if it shouldn't be: `shared_ptr<seastar::tls::server_credentials> lc`
> But we can leave that to them to decide.

I think that `listen_options` are generic for all `server_socket` types.

**Pull request closed without merging**